test/mitmproxy/proxy2/layers/http/test_http_fuzz.py

```diff
@@ -0,0 +1,400 @@
+from typing import Tuple, Dict, Any
+
+import pytest
+from h2.settings import SettingCodes
+from hypothesis import example, given
+from hypothesis.strategies import binary, booleans, composite, dictionaries, integers, lists, sampled_from, sets, text, \
+    data
+
+from mitmproxy import options
+from mitmproxy.addons.proxyserver import Proxyserver
+from mitmproxy.http import HTTPFlow
+from mitmproxy.proxy.protocol.http import HTTPMode
+from mitmproxy.proxy2 import context, events
+from mitmproxy.proxy2.commands import OpenConnection, SendData
+from mitmproxy.proxy2.context import Server
+from mitmproxy.proxy2.events import DataReceived, Start, ConnectionClosed
+from mitmproxy.proxy2.layers import http
+from test.mitmproxy.proxy2.layers.http.hyper_h2_test_helpers import FrameFactory
+from test.mitmproxy.proxy2.layers.http.test_http2 import make_h2, example_response_headers, example_request_headers, \
+    start_h2_client
+from test.mitmproxy.proxy2.tutils import Placeholder, Playbook, reply, _TracebackInPlaybook, _eq
+
+opts = options.Options()
+Proxyserver().load(opts)
+
+request_lines = sampled_from([
+    b"GET / HTTP/1.1",
+    b"GET http://example.com/ HTTP/1.1",
+    b"CONNECT example.com:443 HTTP/1.1",
+    b"HEAD /foo HTTP/0.9",
```

Comment:
These HTTP/0.9 request lines are a bit nonsensical, actual HTTP/0.9 requests only support GET and look like:
```
GET /foo
```

Does mitmproxy actually support HTTP/0.9 now? I didn't [include a `request.is_http09` property](https://github.com/mitmproxy/mitmproxy/blob/b7efe9b2d4b986933f904912324b770dfb3e3da4/mitmproxy/net/http/message.py#L73-L83) because it didn't appear to be supported.

Code later on appears to assume that a version is always present, but with `HTTP/0.9` the version is missing from the request. So I don't think HTTP/0.9 is supported.

Reply:
We don't support HTTP/0.9, I was just making up values for fuzzing here. To be honest I wasn't aware that HTTP/0.9 doesn't always include the version, the more you learn... 🙂